src/main/java/mate/academy/bookstore/serivce/impl/BookServiceImpl.java

```diff
@@ -11,14 +15,26 @@
 @RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
     }
 
     @Override
-    public List<Book> finadAll() {
-        return bookRepository.findAll();
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book by id" + id)
+        );
+        return bookMapper.toDto(book);
```

Comment:
a small but pretty suggestion 😉 
```suggestion
        return bookRepository.findById(id)
                .map(bookMapper::toDto)
                .orElseThrow(
                    () -> new EntityNotFoundException("Can't find book by id" + id));
```